examples/header_bar/title_bar.rs

```diff
@@ -0,0 +1,27 @@
+use gtk::{prelude::*, Label};
+use gtk::{traits::StackSwitcherExt, HeaderBar, StackSwitcher};
+
+pub struct TitleBar {
+    header: HeaderBar,
+    menu: StackSwitcher,
+}
+
+impl TitleBar {
+    pub fn new() -> Self {
+        let menu = StackSwitcher::new();
+        let header = HeaderBar::builder()
+            .custom_title(&menu)
+            .show_close_button(true)
+            .build();
+        header.add(&Label::new(Some("Custom title")));
+        Self { header, menu }
+    }
+
+    pub fn set_stack(&self, stack: gtk::Stack) {
```

Comment:
Why take this by value if you only need a ref?

Reply:
I don't think I considered the life time of ref. I modified it.